# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGL2Context.cpp
# Commit: 71c122ac0ca7
# Full Hash: 71c122ac0ca73391866b1ef19f4f82bc2d28568b
# Author: David Parks <davidp99@gmail.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Initial out-of-process WebGL implementation. r=mccr8,handyman
#   
#   Splits WebGLContext into ClientWebGLContext and HostWebGLContext.  The Client enables the JS-control of a WebGL context in a content procecss while the Host executes the WebGL graphics operations (via a WebGLContext that maintains much of the existing code) in the compositor process.  At this point, the cross-process behavior is disabled -- this series of patches is an incremental step toward that final goal.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D54018
# ==============================================================================

diff -r 5cbf5bb11c58 -r 71c122ac0ca7 dom/canvas/WebGL2Context.cpp
--- a/dom/canvas/WebGL2Context.cpp	Wed Jan 08 21:52:03 2020 +0000
+++ b/dom/canvas/WebGL2Context.cpp	Wed Jan 08 22:19:14 2020 +0000
@@ -17,14 +17,6 @@
 
 namespace mozilla {
 
-WebGL2Context::WebGL2Context() : WebGLContext() {
-  MOZ_ASSERT(IsSupported(),
-             "not supposed to create a WebGL2Context"
-             "context when not supported");
-}
-
-WebGL2Context::~WebGL2Context() {}
-
 UniquePtr<webgl::FormatUsageAuthority> WebGL2Context::CreateFormatUsage(
     gl::GLContext* gl) const {
   return webgl::FormatUsageAuthority::CreateForWebGL2(gl);
@@ -33,14 +25,6 @@
 /*static*/
 bool WebGL2Context::IsSupported() { return StaticPrefs::webgl_enable_webgl2(); }
 
-/*static*/
-WebGL2Context* WebGL2Context::Create() { return new WebGL2Context(); }
-
-JSObject* WebGL2Context::WrapObject(JSContext* cx,
-                                    JS::Handle<JSObject*> givenProto) {
-  return dom::WebGL2RenderingContext_Binding::Wrap(cx, this, givenProto);
-}
-
 ////////////////////////////////////////////////////////////////////////////////
 // WebGL 2 initialisation
 